False positive `unused-variable` for `TYPE_CHECKING` imports

### Bug description

When `allow-global-unused-variables` is set to `false` in the configuration, the `unused-variable` warning is raised for imports made under `TYPE_CHECKING` guard clauses. This behavior seems inconsistent with `unused-import`, which is not raised for such cases.

```
# test.py
# pylint: disable=missing-module-docstring

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import math  # [unused-variable]
``` 

### Configuration

```ini
[tool.pylint.variables]
allow-global-unused-variables=false
```


### Command used

```shell
pylint test.py
```


### Pylint output

```shell
************* Module test
test.py:6:4: W0612: Unused variable 'math' (unused-variable)
```


### Expected behavior

No warnings raised.

### Pylint version

```shell
pylint 3.0.0b1
astroid 3.0.0a2
Python 3.8.10 (default, Nov 14 2022, 12:59:47)
[GCC 9.4.0]
```


### OS / Environment

WSL2 Ubuntu 20.04 LTS

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8696.false_positive
+++ b/doc/whatsnew/fragments/8696.false_positive
@@ -1,0 +1,3 @@
+Fix a false positive for ``unused-variable`` when there is an import in a ``if TYPE_CHECKING:`` block and ``allow-global-unused-variables`` is set to ``no`` in the configuration.
+
+Closes #8696

--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -3101,6 +3101,8 @@ class VariablesChecker(BaseChecker):
             return
         for name, node_lst in not_consumed.items():
             for node in node_lst:
+                if in_type_checking_block(node):
+                    continue
                 self.add_message("unused-variable", args=(name,), node=node)
 
     # pylint: disable = too-many-branches

--- a/tests/functional/u/unused/unused_global_variable2.py
+++ b/tests/functional/u/unused/unused_global_variable2.py
@@ -1,2 +1,11 @@
 # pylint: disable=missing-docstring
+
+
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    import math
+
+
 VAR = 'pylint'  # [unused-variable]

--- a/tests/functional/u/unused/unused_global_variable2.txt
+++ b/tests/functional/u/unused/unused_global_variable2.txt
@@ -1,1 +1,1 @@
-unused-variable:2:0:2:3::Unused variable 'VAR':UNDEFINED
+unused-variable:11:0:11:3::Unused variable 'VAR':UNDEFINED

